kimchi/src/circuits/lookup/lookups.rs

```diff
@@ -263,13 +261,30 @@ impl<F: Copy> JointLookup<SingleLookup<F>, LookupTableID> {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize)]
 pub enum LookupPattern {
     ChaCha,
     ChaChaFinal,
     LookupGate,
 }
 
 impl LookupPattern {
+    pub fn max_lookups_per_row(&self) -> usize {
+        match self {
+            LookupPattern::ChaCha => 4,
+            LookupPattern::ChaChaFinal => 4,
+            LookupPattern::LookupGate => 3,
+        }
+    }
+
+    pub fn max_joint_size(&self) -> u32 {
```

Comment:
missing doc, suggest "the maximum number of columns used by the tables of this lookup pattern"